src/main/java/slimeknights/tconstruct/smeltery/client/GuiSearedFurnace.java

```diff
@@ -0,0 +1,130 @@
+package slimeknights.tconstruct.smeltery.client;
+
+import java.util.List;
+import com.google.common.collect.Lists;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.fluids.FluidStack;
+import slimeknights.mantle.client.gui.GuiElement;
+import slimeknights.mantle.client.gui.GuiElementScalable;
+import slimeknights.mantle.client.gui.GuiMultiModule;
+import slimeknights.tconstruct.library.Util;
+import slimeknights.tconstruct.library.client.RenderUtil;
+import slimeknights.tconstruct.smeltery.client.module.GuiSearedFurnaceSideInventory;
+import slimeknights.tconstruct.smeltery.inventory.ContainerSearedFurnace;
+import slimeknights.tconstruct.smeltery.tileentity.TileHeatingStructureFuelTank;
+import slimeknights.tconstruct.smeltery.tileentity.TileSearedFurnace;
+import slimeknights.tconstruct.tools.inventory.ContainerSideInventory;
+
+public class GuiSearedFurnace extends GuiMultiModule {
+
+  public static final ResourceLocation BACKGROUND = Util.getResource("textures/gui/seared_furnace.png");
+  
+  protected GuiElement flame = new GuiElementScalable(176, 76, 28, 28, 256, 256);
+
+  protected final GuiSearedFurnaceSideInventory sideinventory;
+  protected final TileSearedFurnace furnace;
+  private TileHeatingStructureFuelTank.FuelInfo fuelInfo;
+
+  public GuiSearedFurnace(ContainerSearedFurnace container, TileSearedFurnace tile) {
+    super(container);
+    
+    this.furnace = tile;
+
+    sideinventory = new GuiSearedFurnaceSideInventory(this, container.getSubContainer(ContainerSideInventory.class),
+                                                      furnace, furnace.getSizeInventory(), container.calcColumns());
+    addModule(sideinventory);
+  }
+  
+  @Override
+  public void updateScreen() {
+    super.updateScreen();
+
+    // furnace size changed
+    if(furnace.getSizeInventory() != sideinventory.inventorySlots.inventorySlots.size()) {
+      // close screen
+      this.mc.thePlayer.closeScreen();
+    }
+  }
+
+  @Override
+  protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY) {
```

Comment:
Lots of duplicate code in here again, see if you can pull stuff in a common class for smeltery and furnace